instrumented-tests/Benchmark/BenchmarkTests/BenchmarkTests.swift

```diff
@@ -6,28 +6,44 @@
 
 import XCTest
 import HTTPServerMock
+@testable import Datadog
 
 struct ServerConnectionError: Error {
     let description: String
 }
 
+/// Shared server instance for all test cases.
+private(set) var server: ServerMock! // swiftlint:disable:this implicitly_unwrapped_optional
+/// Shared server session for all test cases.
+private(set) var serverSession: ServerSession! // swiftlint:disable:this implicitly_unwrapped_optional
+
 /// Base class providing mock server instrumentation.
 class BenchmarkTests: XCTestCase {
-    var server: ServerMock! // swiftlint:disable:this implicitly_unwrapped_optional
+    override class func setUp() {
+        super.setUp()
+        if server == nil { server = try! setUpMockServerConnection() }
+        if serverSession == nil { serverSession = server.obtainUniqueRecordingSession() }
+    }
 
     override func setUp() {
         super.setUp()
-        server = try! connectToServer()
+        Datadog.initialize(
+            appContext: Datadog.AppContext(mainBundle: Bundle.main),
+            configuration: Datadog.Configuration
+                .builderUsing(clientToken: "client-token")
+                .set(logsEndpoint: .custom(url: serverSession.recordingURL.absoluteString))
+                .build()
+        )
     }
 
     override func tearDown() {
-        server = nil
+        try! Datadog.deinitializeOrThrow()
         super.tearDown()
     }
 
     // MARK: - `HTTPServerMock` connection
 
-    func connectToServer() throws -> ServerMock {
+    private static func setUpMockServerConnection() throws -> ServerMock {
         let testsBundle = Bundle(for: BenchmarkTests.self)
         guard let serverAddress = testsBundle.object(forInfoDictionaryKey: "MockServerAddress") as? String else {
```

Comment:
I missed the part where you get the value from Info.plist...can you point me to this ?

Reply:
Yah, it's quite complex to follow due to lack of fully programmatic build system in Xcode. It's composed of different IDE capabilities.

It works this:
* The "Build Pre-action" (defined [here](https://github.com/DataDog/dd-sdk-ios/pull/35/files#diff-a6d4b9747b42e7454a9ea4407e4ea59a)) executes [generate-http-server-mock-config.sh](https://github.com/DataDog/dd-sdk-ios/blob/ncreated/RUMM-140-add-performance-benchmarks-part-4/tools/config/generate-http-server-mock-config.sh) script:
<img width="953" alt="Screenshot 2020-03-03 at 09 54 04" src="https://user-images.githubusercontent.com/2358722/75758959-0ebae280-5d35-11ea-83fd-dc712be6fc19.png">

* This script writes server IP address to `http-server-mock.xcconfig` which is [.gitignored](https://github.com/DataDog/dd-sdk-ios/pull/35/files#diff-12a0eaaff469bf6c651883ed1a91903c). The `http-server-mock.xcconfig` looks this:
```
MOCK_SERVER_ADDRESS=127.0.0.1:8000
```
* `http-server-mock.xcconfig` is imported conditionally by [`tests-specific-config.xcconfig`](https://github.com/DataDog/dd-sdk-ios/pull/35/files#diff-9010028a8f7408ef36a5c8c02685cb2e).
* `tests-specific-config.xcconfig` is used to provide build-time variables to Xcode build system:
<img width="704" alt="Screenshot 2020-03-03 at 10 00 23" src="https://user-images.githubusercontent.com/2358722/75759478-ebdcfe00-5d35-11ea-80fc-f0bd01f4aa55.png">

* Finally, the `$(MOCK_SERVER_ADDRESS)` variable is used in `Info.plist` dictionary:

<img width="854" alt="Screenshot 2020-03-03 at 10 01 56" src="https://user-images.githubusercontent.com/2358722/75759532-07e09f80-5d36-11ea-9ca8-23326bcf5431.png">

So, for each build:
* The IP is generated and written to `xcconfig`;
* From `xcconfig` it is passed to `Info.plist`;
* It is read from `Info.plist` at runtime.